Source/JavaScriptCore/runtime/FunctionConstructor.cpp

```diff
@@ -146,6 +146,28 @@ JSObject* constructFunction(JSGlobalObject* globalObject, const ArgList& args, c
     auto code = stringifyFunction(globalObject, args, functionName, functionConstructionMode, scope, functionConstructorParametersEndPosition);
     EXCEPTION_ASSERT(!!scope.exception() == code.isNull());
 
+    if (Options::useTrustedTypes() && globalObject->requiresTrustedTypes()) {
+        bool isTrusted = true;
+        auto* structure = globalObject->trustedScriptStructure();
+        if (!structure) {
+            structure = globalObject->globalObjectMethodTable()->trustedScriptStructure(globalObject);
```

Comment:
I think it is fine if we can't freeze the structure, as long as the slow canCompileStrings method can handle the case when you pass in a modified trusted script object too. If we discover that it is common to polyfill this type in the future, then we can replace this with a flag on structure.

Reply:
Right now it won't allow you to eval a modified trusted script object.
I could keep codeForEval as a slow path but that would make eval of any object slower, which seems bad.

If we can freeze the object during creation that solves that problem and that feels like the correct API design to me.